packages/apps/tools/src/components/Global/ModuleExplorer/tabs.tsx

```diff
@@ -0,0 +1,205 @@
+  onModuleChange,
+  onModuleTabClose,
+  onChainTabClose,
+}) => {
+  const map = useMemo(() => modulesToMap(openedModules), [openedModules]);
+
+  const tabs = useMemo(() => mapToTabs(map), [map]);
+
+  const onParentChange = useCallback(
+    (parentKey: Key) => {
+      const activeModuleToSet = map.get(parentKey as string);
+
+      if (activeModuleToSet) {
+        onModuleChange(activeModuleToSet[0]);
+      }
+    },
+    [map, onModuleChange],
+  );
+
+  const onParentClose = useCallback(
+    (
+      item: ITabNode<{
+        title: string;
+        children: ModuleModel[];
+      }>,
+    ) => {
+      const modulesToClose = map.get(item.key as string);
+
+      if (modulesToClose) {
+        const activeModuleToSet = map.get(
+          (item.prevKey ?? item.nextKey) as string,
+        );
+
+        if (activeModuleToSet) {
+          onModuleChange(activeModuleToSet[0]);
+        }
+        onModuleTabClose(modulesToClose);
+      }
+    },
+    [map, onModuleTabClose, onModuleChange],
+  );
+
+  const onModuleChangeInternal = useCallback(
+    (moduleKey: Key) => {
+      const modulish = tabIdToModule(moduleKey as string);
+
+      const activeModuleToSet = map.get(modulish.name)?.find((module) => {
+        return (
+          module.chainId === modulish.chainId &&
+          module.networkId === modulish.networkId
+        );
+      });
+
+      if (activeModuleToSet) {
+        onModuleChange(activeModuleToSet);
+      }
+    },
+    [map, onModuleChange],
+  );
+
+  const onModuleClose = useCallback(
+    (item: ITabNode<ModuleModel>) => {
+      if (map.get(item.value!.name)?.length === 1) {
+        const currIndex = [...map].findIndex(([key, value]) => {
+          return value.includes(item.value!);
+        });
+
+        const previousEntry = [...map][currIndex - 1];
+
+        if (previousEntry) {
+          onModuleChange(previousEntry[1][0]);
+        }
+      }
+
+      onChainTabClose(item.value!);
+    },
+    [map, onChainTabClose, onModuleChange],
+  );
+
+  return (
+    <>
+      <Box
+        paddingBlockStart={'md'}
+        backgroundColor="surfaceHighContrast.default"
+      >
+        <Tabs
+          items={tabs}
+          selectedKey={activeModule.name}
+          onSelectionChange={onParentChange}
+          onClose={onParentClose}
+          tabPanelClassName={firstLevelTabPanelStyles}
+          isCompact
+        >
+          {(item) => (
+            <TabItem key={item.title} title={item.title}>
+              {/* We render nothing, since we don't want to rerender the sub tabs below every single time. */}
+              {null}
+            </TabItem>
+          )}
+        </Tabs>
+      </Box>
+      <Stack
+        alignItems="center"
+        paddingInlineStart={'md'}
+        paddingBlockStart={'md'}
+        backgroundColor="surface.default"
+        className={secondLevelTabContainerStyles}
+      >
+        <Text className={tabsLabelStyles} size="smallest" bold>
+          Module on chain:
```

Comment:
Do we need to translate this one?

Reply:
Good catch, made it use a translation entry now